causalpy/pymc_experiments.py

```diff
@@ -1477,3 +1480,378 @@ def get_naive_OLS_fit(self):
         beta_params.insert(0, ols_reg.intercept_[0])
         self.ols_beta_params = dict(zip(self._x_design_info.column_names, beta_params))
         self.ols_reg = ols_reg
+
+
+class InversePropensityWeighting(ExperimentalDesign, PropensityDataValidator):
+    """
+    A class to analyse inverse propensity weighting experiments.
+
+    :param data:
+        A pandas dataframe
+    :param formula:
+        A statistical model formula for the propensity model
+    :param outcome_variable
+        A string denoting the outcome variable in datq to be reweighted
+    :param weighting_scheme:
+        A string denoting which weighting scheme to use among: 'raw', 'robust',
+        'doubly robust'
+    :param model:
+        A PyMC model
+
+    Example
+    --------
+    >>> import causalpy as cp
+    >>> df = cp.load_data("nhefs")
+    >>> seed = 42
+    >>> result = cp.pymc_experiments.InversePropensityWeighting(
+    ...     df,
+    ...     formula="trt ~ 1 + age + race",
+    ...     outcome_variable ="outcome",
+    ...     weighting_scheme="robust",
+    ...     model=cp.pymc_models.PropensityScore(
+    ...         sample_kwargs={
+    ...             "draws": 100,
+    ...             "target_accept": 0.95,
+    ...             "random_seed": seed,
+    ...             "progressbar": False,
+    ...         },
+    ...     ),
+    ... )
+    """
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        formula: str,
+        outcome_variable: str,
+        weighting_scheme: str,
+        model=None,
+        **kwargs,
+    ):
+        super().__init__(model=model, **kwargs)
+        self.expt_type = "Inverse Propensity Score Weighting"
+        self.data = data
+        self.formula = formula
+        self.outcome_variable = outcome_variable
+        self.weighting_scheme = weighting_scheme
+        self._input_validation()
+
+        t, X = dmatrices(formula, self.data)
+        self._t_design_info = t.design_info
+        self._t_design_info = X.design_info
+        self.labels = X.design_info.column_names
+        self.t, self.X = np.asarray(t), np.asarray(X)
+        self.y = self.data[self.outcome_variable]
+
+        COORDS = {"obs_ind": list(range(self.X.shape[0])), "coeffs": self.labels}
+        self.coords = COORDS
+        self.model.fit(X=self.X, t=self.t, coords=COORDS)
+
+    def make_robust_adjustments(self, ps):
```

Comment:
Can you add a source on how to compute the different adjustement schemes to the docstrings? This will help anybody having to understand that code